pkg/cluster/controllers/storage/healthchecker.go

```diff
@@ -42,19 +42,29 @@ func (c *HealthChecker) CheckHealth(ctx context.Context, svc *lcm.ServiceConfig,
 		o(checkOpts)
 	}
 
+	var res *lcm.CheckResponse
+
+	var err error
+
 	switch checkOpts.StorageDriver {
 	case goharborv1.S3DriverName:
-		return S3StorageHealthCheck(ctx, svc, checkOpts)
+		res, err = S3StorageHealthCheck(ctx, svc, checkOpts)
```

Comment:
@wangcanfengxs 

I suggest wrapping the health check error where the health check func of the specific service health checker is called. Most places are in the `Apply()` method of the service manager. And then you don't need to do changes to the internal logic of the health checker.

e.g: path is `controllers/goharbor/harborcluster/service_mgr.go` 

```go
# line 165
res, err := s.ctrl.HealthChecker().CheckHealth(s.ctx, svcCfg, options...)
		if err != nil {
```